src/Tribe/Views/V2/View.php

```diff
@@ -2194,50 +2194,77 @@ protected function should_show_recent_past_events_view() {
 	 */
 	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
 
-		$manager = tribe( Manager::class );
+		$manager      = tribe( Manager::class );
 		$default_slug = $manager->get_default_view_option();
 
-		// Only recent Past Events to only the default view.
+		// Show Latest Past Events only on the default view.
 		if ( $this->get_slug() !== $default_slug ) {
 			return;
 		}
 
+		$now    = $this->context->get( 'now' );
+		$latest = tribe_events_latest_date();
+
+		// If now is less then the latest event published, do not show.
+		if ( $now < $latest ) {
+			return;
+		}
+
+
+		// Checks to verify on the initial load of a view or if using today's date for the view.
+		$today     = $this->context->get( 'today' );
+		$view_date = $this->context->get( 'event_date', '' );
+		if ( 'month' === $this->get_slug() ) {
+			$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBYEARMONTHTIMEFORMAT );
+			$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBYEARMONTHTIMEFORMAT );
+		} elseif ( 'week' === $this->get_slug() ) {
+			list( $today_week_start, $today_week_end ) = Dates::get_week_start_end( $today, (int) $this->context->get( 'start_of_week', 0 ) );
+			list( $view_week_start, $view_week_end ) = Dates::get_week_start_end( $view_date, (int) $this->context->get( 'start_of_week', 0 ) );
+			$today_formatted     = $today_week_start->format( Dates::DBDATEFORMAT );
+			$view_date_formatted = $view_week_start->format( Dates::DBDATEFORMAT );
+		} else {
+			$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBDATEFORMAT );
+			$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBDATEFORMAT );
+		}
+
+		// If view date is not empty and today does not equal the view date, then do not show.
+		if ( ! empty( $view_date ) && $today_formatted !== $view_date_formatted ) {
+			return;
+		}
+
 		// Flatten Month and Week View Events to Get Accurate Count
-/*		$views_to_flatten = [
+		$views_to_flatten = [
```

Comment:
should this be filtered so that ECP can hook into and do its stuff there?

Reply:
See https://github.com/moderntribe/the-events-calendar/pull/3134#discussion_r407981179 -- I think this block could be removed completely and make this let go of specificity.